### DISCUSSION TITLE
"On the fly" jobs with Quartz.Extensions.Hosting

### DISCUSSION THREAD

**Discussion by mmoore99** [quartznet/quartznet#1503] [Q&A]
All of the examples in the documentation for `Quartz.Extensions.Hosting` show the configuring of all of the jobs and adding them to the DI container at program startup. It is not at all clear how to add and delete jobs dynamically during the execution of the program. Any suggestions or examples?

**Comment by lahma** [comment]
Well you can do it dynamically by getting the `IScheduler` instance. So DI should provide you with `ISchedulerFactory`, get that and call `GetScheduler()` to get the actual instance. Then all the examples of scheduling via `IScheduler` apply.

You can also use the XML scheduling configuration if you configure it to polling the file and updating on changes.

**Reply by mmoore99** [reply]
Thanks for getting back to me. Is there an example showing how to get IScheduleFactory using DI?  I am new to the DI world and it's not all clear yet.

**Reply by lahma** [reply]
If you look at the worker example project in Quartz repo, it's enough to add `ISchedulerFactory` as constructor parameter. Might varmaan based on your use case though.
